enhancements/kube-apiserver/api-webhook-supportability.md

```diff
@@ -0,0 +1,506 @@
+- an **extension of OpenShift enhancement template** about webhooks, their impact to the cluster,
+  possible debugging strategies and supportability concerns
+- a **monitoring dashboard** to watch the latency of individual webhooks and total latency for all
+  webhooks per resource.
+- to **skip 3rd-party admission webhooks on runlevel=0 namespaces** by default, with an explicit
+  opt-in per admission webhook to bypass the restriction.
+
+This enhancement is about admission webhook and conversion webhook. They have an overlap of
+characteristic, but also distinct threat for a cluster.
+
+## Motivation
+
+As seen in escalations of early adopters of new technology on-top of OpenShift like service meshes
+or policy engines like OPA, we know that clusters with webhooks installed for core resources like
+pods or `SubjectAccessReview` are at risk and a supportability problem. The triage is hard or
+impossible if these are installed, and often the webhooks misbehavour is even the root cause for the
+problems. Examples are:
+
+https://bugzilla.redhat.com/show_bug.cgi?id=1998436
+https://bugzilla.redhat.com/show_bug.cgi?id=1998245
+
+Hence, we have learned that API extension webhooks can have negative impact on
+- the **performance** of a cluster because API requests involve additional network round-trips to
+  another service, or a webhook with side-effect causes a reevaluation of the admission chain, or
+  intermediate requests like `SubjectAccessReviews` require yet another webhook call.
+- the **stability** of a cluster because API requests can fail either because the webhook
+  intentionally rejects requests, or the webhook request fails for infrastructure reasons (and
+  failure policy `Fail` is used), or the webhook times out requests (because it is overloaded or for
+  infrastructure reasons)
+- the **security** of a cluster because API requests can include security sensitive information like
+  tokens or secrets. The webhook will see these in plain-text and hence become a security risk for
+  the cluster.
+- the **availability** of a cluster because API requests that fail can render resource-agnostic
+  controllers like garbage collection, namespace controller or resource quota disfunctional for all
+  resources in a cluster, not only those the webhook is about.
+
+### Goals
+
+- allow the customer, support and engineering in escalations to notice misbehaving webhooks
+- make the customer aware about risks of installing (multiple) add-ons that hook deeply into core
+  components
+- make Red Hat teams aware early about the risk of hooking into core resources and make them design
+  their architecture with minimal impact
+- establish an Red Hat approval process for webhooks hooking into core resources
+- establish a culture of "webhook impact budget" in the development process.
+
+### Non-Goals
+
+- tainting a cluster with effect on the support status
+- forbidding webhooks for OpenShift add-ons
```

Comment:
maybe phrase this as "discourage usage of web hooks in cluster" to address Jeremy comment and make it clear that this enhancement is not about telling people "do not use webhooks, they are bad".